Mathlib/LinearAlgebra/Dual/BaseChange.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+module
+
+public import Mathlib.LinearAlgebra.Dual.Defs
+public import Mathlib.LinearAlgebra.FreeModule.Finite.Basic
+public import Mathlib.RingTheory.TensorProduct.IsBaseChangeFree
+public import Mathlib.RingTheory.TensorProduct.IsBaseChangeHom
+/-!
+# Base change for the dual of a module
+
+* `Module.Dual.congr` : equivalent modules have equivalent duals.
+
+If `f : Module.Dual R V` and `Algebra R A`, then
+
+* `Module.Dual.baseChange A f` is the element
+of `Module.Dual A (A ⊗[R] V)` deduced by base change.
+
+* `Module.Dual.baseChangeHom` is the `R`-linear map
+given by `Module.Dual.baseChange`.
+
+* `IsBaseChange.dual` : for finite free modules, taking dual commutes with base change.
+
+## TODO
+
+Generalize for more general modules of linear maps.
+
+-/
+
+@[expose] public section
+
+universe u v
+
+namespace Module.Dual
+
+open TensorProduct LinearEquiv
+
+variable {R : Type*} [CommSemiring R]
+  {V : Type*} [AddCommMonoid V] [Module R V]
+  {W : Type*} [AddCommMonoid W] [Module R W]
+  (A : Type*) [CommSemiring A] [Algebra R A]
+
+/-- Equivalent modules have equivalent duals. -/
+def congr (e : V ≃ₗ[R] W) :
+    Module.Dual R V ≃ₗ[R] Module.Dual R W :=
+  LinearEquiv.congrLeft R R e
+
+/-- `LinearMap.baseChange` for `Module.Dual`. -/
+def baseChange : Module.Dual R V →ₗ[R] Module.Dual A (A ⊗[R] V) :=
+  (AlgebraTensorModule.rid R A A).compRight R ∘ₗ LinearMap.baseChangeHom R A V R
+
+@[simp]
+theorem baseChange_apply_tmul (f : Module.Dual R V) (a : A) (v : V) :
+    f.baseChange A (a ⊗ₜ v) = (f v) • a :=
+  rfl
+
+variable {B : Type*} [CommSemiring B] [Algebra R B] [Algebra A B] [IsScalarTower R A B]
+
+theorem baseChange_baseChange (f : Module.Dual R V) :
+    ((f.baseChange A).baseChange B) =
+      (congr (TensorProduct.AlgebraTensorModule.cancelBaseChange R A B B V)).symm
+       (f.baseChange B) := by
+  ext; simp [congr, congrLeft]
```

Comment:
I suggest that we add `@[simps]` to `LinearEquiv.congrLeft` [here](https://github.com/leanprover-community/mathlib4/blob/7c89bb6086e31db7f332c374f780cac5416ef641/Mathlib/Algebra/Module/Equiv/Basic.lean#L724) and then make the following style tweaks here:
```suggestion
open AlgebraTensorModule in
theorem baseChange_baseChange (f : Module.Dual R V) :
    (f.baseChange A).baseChange B = (congr (cancelBaseChange R A B B V)).symm (f.baseChange B) := by
  ext; simp
```